File: backend/src/main/java/com/endfiled/model/Character.java
package com.endfiled.model;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("`character`")
public class Character {
    @TableId
    private String id;
    private String name;
    private String icon;
    private Integer rarity;
    private Integer level;
    private BigDecimal baseHp;
    private BigDecimal baseAtk;
    private BigDecimal baseStr;
    private BigDecimal baseAgi;
    private BigDecimal baseInt;
    private BigDecimal baseWil;
    private String mainAttr;
    private String subAttr;
    private String profession;
    private String element;
    private String weaponType;
    private Integer potential;
    private Integer trustTalent;
    private String reserve1;
    private String reserve2;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/engine/types/character.ts
export type Rarity = 4 | 5 | 6;
export type Profession = 'assault' | 'guard' | 'caster' | 'heavy' | 'vanguard' | 'support';
export type ElementType = 'pyro' | 'cryo' | 'electro' | 'natural' | 'physical';
export type WeaponType = 'sword' | 'greatsword' | 'polearm' | 'pistol' | 'caster_unit';
export type AttrType = 'str' | 'agi' | 'int' | 'wil';

export interface Character {
  id: string;
  name: string;
  icon: string;
  rarity: Rarity;
  level: number;
  baseHp: number;
  baseAtk: number;
  baseStr: number;
  baseAgi: number;
  baseInt: number;
  baseWil: number;
  mainAttr: AttrType;
  subAttr: AttrType;
  profession: Profession;
  element: ElementType;
  weaponType: WeaponType;
  potential: number;
}
